boost::python::make_setter(&X::y) no longer compiles

This trivial example, similar to the examples in the Boost.Python documentation, compiled with 1.58 but doesn't with 1.59 (using various versions of GCC and Clang):

    #include <boost/python.hpp>

    struct X { int y; };

    int main()
    {
      boost::python::make_setter(&X::y);
    }


The relevant error is:

`/usr/local/boost-1.59.0/include/boost/python/data_members.hpp:303:15: note: candidate: boost::python::api::object boost::python::make_setter(D&) [with D = int X::*] <near match>`
<pre>    inline object make_setter(D& x)
                  ^</pre>
`/usr/local/boost-1.59.0/include/boost/python/data_members.hpp:303:15: note:   conversion of argument 1 would be ill-formed:`<BR>
`prog.cc:7:37: error: invalid initialization of non-const reference of type 'int X::*&' from an rvalue of type 'int X::*'`
<pre>     boost::python::make_setter(&X::y);
                                     ^</pre>

--- a/include/boost/python/data_members.hpp
+++ b/include/boost/python/data_members.hpp
@@ -305,7 +305,7 @@ inline object make_setter(D& x)
     return detail::make_setter(x, default_call_policies(), is_member_pointer<D>(), 0);
 }
 
-# if BOOST_WORKAROUND(__EDG_VERSION__, <= 238)
+# if !BOOST_WORKAROUND(__EDG_VERSION__, <= 238)
 template <class D>
 inline object make_setter(D const& x)
 {

